pkg/virt-handler/spice-proxy/spice-proxy.go

```diff
@@ -0,0 +1,224 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2019 Red Hat, Inc.
+ *
+ */
+package spice_proxy
+
+//go:generate mockgen -source $GOFILE -package=$GOPACKAGE -destination=generated_mock_$GOFILE
+
+/*
+ ATTENTION: Rerun code generators when interface signatures are modified.
+*/
+
+import (
+	"fmt"
+	"io"
+	"net"
+	"sync"
+
+	v1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/client-go/log"
+	"kubevirt.io/kubevirt/pkg/virt-handler/isolation"
+)
+
+var (
+	proxyRange = []int{30000, 32000}
```

Comment:
Are we sure 2000 connections will be enough? Not sure about the scale we aim for.

Reply:
`2000` ports mean 2000 vms on the same host I can of course change it to higher number if required.

maybe make it configurable via the virt-config?

Reply:
Couldn't the handler already use a websocket mechanism?
WOuldn#t this allow us to open just a single port?